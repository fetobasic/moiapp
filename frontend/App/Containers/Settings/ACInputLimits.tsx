import React, { useMemo } from 'react';
import { View, StyleSheet, Text, ScrollView } from 'react-native';
import { ApplicationStyles, Colors, Fonts, Metrics } from 'App/Themes';
import { HeaderSimple as Header, InformationRow, Pressable, SectionWithTitle } from 'App/Components';
import renderElement from 'App/Services/RenderElement';
import Information from 'App/Images/Icons/information.svg';
import Row from 'App/Components/Pairing/InfoRow';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { SettingsStackParamList } from 'App/Types/NavigationStackParamList';
import { useSelector } from 'react-redux';
import { getCurrentDevice } from 'App/Store/Devices/selectors';
import HelpIcon from 'App/Images/Icons/help.svg';
import SocketIcon from 'App/Images/Icons/socket.svg';
import RvSocketIcon from 'App/Images/Icons/rvSocket.svg';
import { Yeti6GState } from 'App/Types/Yeti';
import { getPowerLimits } from 'App/Services/ACInverter';
import { isYeti20004000 } from 'App/Services/ThingHelper';
import { useAppSelector } from 'App/Hooks';
import { convertTemperature } from 'App/Services/ConvertTemperature';

type Props = NativeStackScreenProps<SettingsStackParamList, 'AcInputLimits'>;

const AcInputLimits = ({ navigation, route }: Props) => {
  const { thingName } = route.params;
  const device = useSelector(getCurrentDevice(thingName || '')) as Yeti6GState;
  const { temperature } = useAppSelector((state) => ({
    temperature: state.application.units.temperature,
  }));

  const showAdditionalInfo = isYeti20004000(device);
  const { maxLimitA, maxLimitW } = getPowerLimits(device);

  const wattsRate = useMemo(
    () => device?.config?.inv?.v ?? maxLimitW / maxLimitA,
    [device?.config?.inv?.v, maxLimitA, maxLimitW],
  );
  const wallInputPortLimit = useMemo(
    () => device?.config?.ports?.acIn?.wall?.aLmt ?? 0,
    [device?.config?.ports?.acIn?.wall?.aLmt],
  );
  const wallBatteryChargeLimit = useMemo(
    () => device?.config?.ports?.acIn?.wall?.aChgLmt ?? 0,
    [device?.config?.ports?.acIn?.wall?.aChgLmt],
  );
  const plusInputPortLimit = useMemo(
    () => device?.config?.ports?.acIn?.wall?.aLmt ?? 0,
    [device?.config?.ports?.acIn?.wall?.aLmt],
  );
  const plusBatteryChargeLimit = useMemo(
    () => device?.config?.ports?.acIn?.wall?.aChgLmt ?? 0,
    [device?.config?.ports?.acIn?.wall?.aChgLmt],
  );

  const isLowLimit = useMemo(() => !device?.status?.ports?.acIn?.hLmt, [device?.status?.ports?.acIn?.hLmt]);

  return (
    <View style={ApplicationStyles.mainContainer}>
      <Header
        title="AC Inverter/Charger"
        rightSection={renderElement(
          <Pressable onPress={() => navigation.navigate('InputLimitsHelp')}>
            <HelpIcon />
          </Pressable>,
        )}
      />
      <ScrollView style={styles.container}>
        <Row
          containerWBStyle={styles.row}
          withBorder={false}
          icon={renderElement(<Information />)}
          titleTextStyles={styles.textTitle}
          title="If the AC Profile does not match the AC source, AC power pass-through stops and the battery is used as the power source."
        />
        <SectionWithTitle title="AC Profile">
          <InformationRow title="Voltage" description={`${device?.config?.inv?.v ?? 0} V`} style={styles.border} />
          <InformationRow title="Frequency" description={`${device?.config?.inv?.hz ?? 0} Hz`} />
        </SectionWithTitle>

        {showAdditionalInfo && (
          <SectionWithTitle containerStyle={styles.sectionInverterTemperature}>
            <InformationRow
              title="Inverter Temperature"
              description={convertTemperature(temperature, device?.status?.inv?.cTmp ?? 0, false, true)}>
              <Text style={styles.textDisabled}>Power electronics temperature</Text>
            </InformationRow>
          </SectionWithTitle>
        )}

        <SectionWithTitle title="Limits">
          <InformationRow
            title="AC Limit Setting"
            description={isLowLimit ? 'LOW (Custom)' : 'HIGH'}
            descriptionStyle={isLowLimit ? styles.lowLimit : styles.textDisabled}>
            <Text style={styles.textDescription}>
              The AC Limit Setting can only be changed on the Yeti device. The limits can be customized in the app when
              this is set to LOW.
            </Text>
          </InformationRow>
          <InformationRow
            title="AC Charge Port"
            titleStyle={!isLowLimit ? styles.textDisabled : undefined}
            addLeftIcon={<SocketIcon style={styles.socketIcon} color={isLowLimit ? Colors.green : Colors.disabled} />}
            rightIconStyle={!isLowLimit ? styles.sectionDisabled : undefined}
            pressableStyle={!isLowLimit ? styles.pressableDisabled : undefined}
            onPress={() => (isLowLimit ? navigation.navigate('AcChargeInputLimit', { device }) : () => {})}>
            <View>
              <View style={styles.postsDescription}>
                <Text style={[styles.textDescription, ApplicationStyles.flex]}>Input Port Limit</Text>
                <Text style={styles.textDescription}>
                  About {wallInputPortLimit}A / {wattsRate * wallInputPortLimit}W
                </Text>
              </View>
              <View style={styles.postsDescription}>
                <Text style={[styles.textDescription, ApplicationStyles.flex]}>Battery Charge Limit</Text>
                <Text style={styles.textDescription}>
                  About {wallBatteryChargeLimit}A / {wattsRate * wallBatteryChargeLimit}W
                </Text>
              </View>
            </View>
          </InformationRow>
          {showAdditionalInfo && (
            <InformationRow
              title="Power+ Port"
              titleStyle={!isLowLimit ? styles.textDisabled : undefined}
              addLeftIcon={
                <RvSocketIcon style={styles.socketIcon} color={isLowLimit ? Colors.green : Colors.disabled} />
              }
              rightIconStyle={!isLowLimit ? styles.sectionDisabled : undefined}
              pressableStyle={!isLowLimit ? styles.pressableDisabled : undefined}
              onPress={() => (isLowLimit ? navigation.navigate('PowerInputLimit', { device }) : () => {})}>
              <View>
                <View style={styles.postsDescription}>
                  <Text style={[styles.textDescription, ApplicationStyles.flex]}>Input Port Limit</Text>
                  <Text style={styles.textDescription}>
                    About {plusInputPortLimit}A / {wattsRate * plusInputPortLimit}W
                  </Text>
                </View>
                <View style={styles.postsDescription}>
                  <Text style={[styles.textDescription, ApplicationStyles.flex]}>Battery Charge Limit</Text>
                  <Text style={styles.textDescription}>
                    About {plusBatteryChargeLimit}A / {wattsRate * plusBatteryChargeLimit}W
                  </Text>
                </View>
              </View>
            </InformationRow>
          )}
        </SectionWithTitle>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: Metrics.baseMargin,
    paddingVertical: Metrics.marginSection,
  },
  row: {
    paddingVertical: 2,
  },
  textTitle: {
    ...Fonts.font.base.bodyOne,
  },
  textDescription: {
    ...Fonts.font.base.caption,
    color: Colors.disabled,
  },
  textDisabled: {
    color: Colors.disabled,
  },
  sectionDisabled: {
    opacity: 0.5,
  },
  pressableDisabled: {
    opacity: 1,
  },
  border: {
    borderBottomColor: Colors.border,
    borderBottomWidth: 1,
  },
  lowLimit: {
    color: Colors.green,
  },
  socketIcon: {
    marginLeft: -Metrics.smallMargin,
    marginRight: Metrics.smallMargin,
  },
  postsDescription: {
    flexDirection: 'row',
    marginRight: Metrics.smallMargin,
    marginTop: 2,
  },
  sectionInverterTemperature: {
    paddingTop: 0,
  },
});

export default AcInputLimits;
