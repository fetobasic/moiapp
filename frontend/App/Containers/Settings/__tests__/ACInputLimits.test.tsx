import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { mockedState } from 'App/Fixtures/mocks/mockedState';
import { yeti6GState } from 'App/Fixtures/yeti6GState';
import ACInputLimits from '../ACInputLimits';

const mockStore = createMockStore([]);
const store = mockStore({
  ...mockedState,
});

describe('ACInputLimits', () => {
  test('renders screen', () => {
    const props = {
      navigation: {
        navigate: jest.fn() as any,
        canGoBack: jest.fn() as any,
      },
      route: { params: { thingName: '123', device: yeti6GState } },
    } as any;

    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <ACInputLimits {...props} />
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
