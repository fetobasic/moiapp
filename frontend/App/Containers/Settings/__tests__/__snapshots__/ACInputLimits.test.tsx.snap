// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACInputLimits renders screen 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(27, 27, 27, 1)",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "elevation": 12,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 12,
          "zIndex": 1,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "opacity": 1,
            },
            {
              "left": 0,
              "padding": 10,
              "position": "absolute",
            },
          ]
        }
      >
        <SvgMock />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginHorizontal": 48,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "RobotoCondensed-Medium",
              "fontSize": 20,
              "fontWeight": "500",
              "lineHeight": 32,
            }
          }
        >
          AC Inverter/Charger
        </Text>
      </View>
      <View
        style={
          {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "opacity": 1,
              },
              undefined,
            ]
          }
        >
          <SvgMock />
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      {
        "paddingHorizontal": 24,
        "paddingVertical": 16,
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(27, 27, 27, 1)",
              "borderRadius": 16,
              "paddingHorizontal": 12,
              "paddingVertical": 15,
            },
            undefined,
            {
              "marginBottom": 8,
              "marginHorizontal": 1,
              "paddingVertical": 16,
            },
            {
              "paddingVertical": 2,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "height": 32,
                "justifyContent": "center",
                "width": 32,
              }
            }
          >
            <SvgMock />
          </View>
          <View
            style={
              [
                {
                  "flex": 1,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                },
                undefined,
                {
                  "paddingRight": 0,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginVertical": 4,
                  },
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                  },
                ]
              }
            >
              If the AC Profile does not match the AC source, AC power pass-through stops and the battery is used as the power source.
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "borderColor": "rgba(54, 54, 54, 1)",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginVertical": 12,
              "paddingBottom": 6,
              "paddingHorizontal": 8,
              "paddingTop": 12,
              "position": "relative",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "rgba(27, 27, 27, 1)",
                "color": "rgba(255, 255, 255, 0.7)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 12,
                "fontWeight": "300",
                "left": 8,
                "lineHeight": 16,
                "paddingLeft": 12,
                "position": "absolute",
                "top": -9,
                "width": "100%",
              },
              undefined,
              undefined,
            ]
          }
        >
          AC Profile
        </Text>
        <View
          style={
            [
              {
                "paddingHorizontal": 12,
                "paddingVertical": 15,
              },
              {
                "borderBottomColor": "rgba(54, 54, 54, 1)",
                "borderBottomWidth": 1,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                      "maxWidth": 210,
                    },
                    undefined,
                  ]
                }
              >
                Voltage
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  false,
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "opacity": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="clip"
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "rgba(255, 255, 255, 0.87)",
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                        "maxWidth": 210,
                      },
                      undefined,
                    ]
                  }
                >
                  0 V
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "paddingHorizontal": 12,
                "paddingVertical": 15,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                      "maxWidth": 210,
                    },
                    undefined,
                  ]
                }
              >
                Frequency
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  false,
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "opacity": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="clip"
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "rgba(255, 255, 255, 0.87)",
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                        "maxWidth": 210,
                      },
                      undefined,
                    ]
                  }
                >
                  0 Hz
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "borderColor": "rgba(54, 54, 54, 1)",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginVertical": 12,
              "paddingBottom": 6,
              "paddingHorizontal": 8,
              "paddingTop": 12,
              "position": "relative",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "rgba(27, 27, 27, 1)",
                "color": "rgba(255, 255, 255, 0.7)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 12,
                "fontWeight": "300",
                "left": 8,
                "lineHeight": 16,
                "paddingLeft": 12,
                "position": "absolute",
                "top": -9,
                "width": "100%",
              },
              undefined,
              undefined,
            ]
          }
        >
          Limits
        </Text>
        <View
          style={
            [
              {
                "paddingHorizontal": 12,
                "paddingVertical": 15,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                      "maxWidth": 210,
                    },
                    undefined,
                  ]
                }
              >
                AC Limit Setting
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  false,
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "opacity": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="clip"
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "rgba(255, 255, 255, 0.87)",
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                        "maxWidth": 210,
                      },
                      {
                        "color": "rgba(191, 210, 43, 1)",
                      },
                    ]
                  }
                >
                  LOW (Custom)
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              {
                "color": "rgba(133, 133, 133, 1)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 12,
                "fontWeight": "300",
                "lineHeight": 16,
              }
            }
          >
            The AC Limit Setting can only be changed on the Yeti device. The limits can be customized in the app when this is set to LOW.
          </Text>
        </View>
        <View
          style={
            [
              {
                "borderRadius": 16,
                "elevation": 2,
                "marginBottom": 8,
                "shadowColor": "rgba(0, 0, 0, 0.4)",
                "shadowOffset": {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.7,
                "shadowRadius": 7,
              },
              undefined,
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 16,
                  "marginHorizontal": 1,
                  "overflow": "hidden",
                  "paddingTop": 1,
                  "position": "relative",
                },
                undefined,
                undefined,
              ]
            }
          >
            <RNSVGSvgView
              bbHeight="100%"
              bbWidth="100%"
              focusable={false}
              height="100%"
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              width="100%"
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGDefs>
                  <RNSVGLinearGradient
                    gradient={
                      [
                        0,
                        -13224394,
                        1,
                        1776411,
                      ]
                    }
                    gradientTransform={null}
                    gradientUnits={0}
                    name="grad"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0.5"
                  />
                </RNSVGDefs>
                <RNSVGRect
                  fill={
                    {
                      "brushRef": "grad",
                      "type": 1,
                    }
                  }
                  height="100%"
                  propList={
                    [
                      "fill",
                    ]
                  }
                  width="100%"
                  x="0"
                  y="0"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(27, 27, 27, 1)",
                    "borderRadius": 16,
                    "paddingHorizontal": 12,
                    "paddingVertical": 15,
                  },
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "opacity": 1,
                    },
                    [
                      {
                        "zIndex": 1000,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
                testID="withPressable"
              >
                <View
                  style={
                    [
                      {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "marginLeft": 8,
                        },
                        undefined,
                      ]
                    }
                  >
                    <SvgMock
                      color="rgba(191, 210, 43, 1)"
                      style={
                        {
                          "marginLeft": -8,
                          "marginRight": 8,
                        }
                      }
                    />
                  </View>
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        [
                          {
                            "color": "rgba(255, 255, 255, 0.87)",
                            "fontFamily": "Roboto-Medium",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "lineHeight": 24,
                            "maxWidth": 210,
                          },
                          undefined,
                        ]
                      }
                    >
                      AC Charge Port
                    </Text>
                  </View>
                  <View
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        false,
                      ]
                    }
                  >
                    <SvgMock
                      style={
                        [
                          {
                            "marginLeft": 16,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
                <View>
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "marginRight": 8,
                        "marginTop": 2,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "rgba(133, 133, 133, 1)",
                            "fontFamily": "Roboto-Medium",
                            "fontSize": 12,
                            "fontWeight": "300",
                            "lineHeight": 16,
                          },
                          {
                            "flex": 1,
                          },
                        ]
                      }
                    >
                      Input Port Limit
                    </Text>
                    <Text
                      style={
                        {
                          "color": "rgba(133, 133, 133, 1)",
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 12,
                          "fontWeight": "300",
                          "lineHeight": 16,
                        }
                      }
                    >
                      About 
                      0
                      A / 
                      0
                      W
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "flexDirection": "row",
                        "marginRight": 8,
                        "marginTop": 2,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "color": "rgba(133, 133, 133, 1)",
                            "fontFamily": "Roboto-Medium",
                            "fontSize": 12,
                            "fontWeight": "300",
                            "lineHeight": 16,
                          },
                          {
                            "flex": 1,
                          },
                        ]
                      }
                    >
                      Battery Charge Limit
                    </Text>
                    <Text
                      style={
                        {
                          "color": "rgba(133, 133, 133, 1)",
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 12,
                          "fontWeight": "300",
                          "lineHeight": 16,
                        }
                      }
                    >
                      About 
                      0
                      A / 
                      0
                      W
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
